addons/mail/static/src/components/attachment_image/attachment_image.scss

```diff
@@ -0,0 +1,118 @@
+// ------------------------------------------------------------------
+// Layout
+// ------------------------------------------------------------------
+.o_AttachmentImage {
+    min-width: 50px;
+    min-height: 50px;
+    flex-shrink: 0;
+    margin: map-get($spacers, 1);
+
+    img {
+        object-fit: contain;
+        margin: 0 auto;
+    }
+
+    &.o-details-overlay:hover {
+        .o_AttachmentImage_imageOverlay {
+            opacity: 1;
+        }
+    }
```

Comment:
Would be nice to avoid CSS specificity and pass `o-details-overlay:hover` on root to this part.

This would require modelling hover to do so... It's a bit annoying, and I prefer to have this PR focused on what it does (re-structure components and models) rather than increase its scope and fix SCSS too.

So all-in-all, let's keep it that way for this PR. Comment is just a reminder to fix this in the future.